code/datums/components/heat_sensitive.dm

```diff
@@ -0,0 +1,26 @@
+/datum/component/heat_sensitive
+	var/max_heat
+	var/min_heat
+	var/atom/target
+
+/datum/component/heat_sensitive/Initialize(max, min)
+	if(!isatom(parent)) //How
+		return COMPONENT_INCOMPATIBLE
+	max_heat = max
+	min_heat = min
+	target = get_atom_on_turf(parent)
+	RegisterSignal(target, COMSIG_MOVABLE_MOVED, .proc/reset_targeting) //If our holder moves, move our turf watcher
```

Comment:
What if the atom itself moves out of the holder. That's why we have movement tracker, but it seems it would be pretty expensive to use here. Might be worth making a cheap and precise options here.

Reply:
Should I make a signal that sends whenever an atom is taken in/out of something/teleported/picked up?

Reply:
What if make something like `COMSIG_HOLDER_MOVED` that say to atoms in `holder.contents` that `holder` is moved
so we can hook up signal to `parent` in contents instead of search for atom on turf.

Reply:
that's a very good idea, what do you think of `/datum/movement_detector`

Reply:
actually one sec I'm dumb